src/test/java/com/treasuredata/client/TestTDHttpClient.java

```diff
@@ -78,4 +93,127 @@ public void deleteMethodTest()
             logger.warn("error", e);
         }
     }
+
+    @Test
+    public void retryOn429()
+            throws Exception
+    {
+        // Configure an artificially low retry interval so we can measure with some confidence that Retry-After is respected
+        client = TDClient.newBuilder()
+                .setRetryMaxIntervalMillis(100)
+                .setRetryLimit(1000)
+                .build()
+                .httpClient;
+
+        final AtomicLong firstRequestNanos = new AtomicLong();
+        final AtomicLong secondRequestNanos = new AtomicLong();
+        final AtomicInteger requests = new AtomicInteger();
+
+        final TDApiRequest req = TDApiRequest.Builder.GET("/v3/system/server_status").build();
+        final byte[] body = "foobar".getBytes("UTF-8");
+        final long retryAfterSeconds = 5;
+
+        ContentResponse resp = client.submitRequest(req, Optional.<String>absent(), new TDHttpClient.DefaultContentHandler()
+        {
+            @Override
+            public ContentResponse submit(Request request)
+                    throws InterruptedException, ExecutionException, TimeoutException
+            {
+                List<Response.ResponseListener> listeners = ImmutableList.of();
+                switch (requests.incrementAndGet()) {
+                    case 1: {
+                        firstRequestNanos.set(System.nanoTime());
+                        HttpResponse response = new HttpResponse(request, listeners)
+                                .status(429);
+                        response.getHeaders().add("Retry-After", Long.toString(retryAfterSeconds));
+                        return new HttpContentResponse(response, new byte[] {}, "", "");
+                    }
+                    case 2: {
+                        secondRequestNanos.set(System.nanoTime());
+                        HttpResponse response = new HttpResponse(request, listeners)
+                                .status(200);
+                        return new HttpContentResponse(response, body, "plain/text", "UTF-8");
+                    }
+                    default:
+                        throw new AssertionError();
+                }
+            }
+        });
+
+        assertThat(requests.get(), is(2));
+        assertThat(resp.getStatus(), is(200));
+        assertThat(resp.getContent(), is(body));
+
+        long delayNanos = secondRequestNanos.get() - firstRequestNanos.get();
+        assertThat(delayNanos, Matchers.greaterThanOrEqualTo(SECONDS.toNanos(retryAfterSeconds)));
+    }
+
+    @Test
+    public void failOn429_TimeLimitExceeded()
+            throws Exception
+    {
+        client = TDClient.newBuilder()
+                .setRetryMaxIntervalMillis(1000)
+                .setRetryLimit(3)
+                .build()
+                .httpClient;
+
+        final long retryAfterSeconds = 4711;
```

Comment:
It's just an arbitrary long `Retry-After` value that is greater than the retry limit * interval configured by the user.

The purpose is to verify that the `TDClientHttpTooManyRequestsException` error is propagated immediately instead of performing pointless retries.